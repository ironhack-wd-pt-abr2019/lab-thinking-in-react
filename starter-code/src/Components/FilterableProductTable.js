import React, {useContext} from 'react';
import SearchBoxComponent from './SearchBoxComponent';
import DataTable from './DataTable';




const filterableproducttable = () =>{

  return (
    <div>
      <h3>IronStore</h3>
      <div> 
        <SearchBoxComponent/>
        <DataTable id="dataTable" />
      </div>
    </div>
  
  )
}

export default filterableproducttable;