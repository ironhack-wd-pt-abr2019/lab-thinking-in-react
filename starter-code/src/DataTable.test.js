import React from 'react';
import ReactDOM,{render, unmountComponentAtNode } from 'react-dom';
import pretty from "pretty";
import { act } from "react-dom/test-utils";
import DataTable from './Components/DataTable';
import SearchBoxComponent from './Components/SearchBoxComponent'
import FilterableProductTable from './Components/FilterableProductTable'
import App from './App'
import {LeProvedore, LeContexte } from './Components/dataContext'

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("Prueba del componente de DataTable", ()=>{
    act(()=>{
        render(<LeProvedore>
            <SearchBoxComponent />
            <DataTable /> 
        </LeProvedore>, container)
        //console.log(container.textContent)
    })
    expect(container.textContent).toMatch(/Name/)
        
})

it("Que le pueda dar click al control del checkbox", ()=>{
  const onChange = jest.fn()
  act(()=>{
    render(<LeProvedore>
      <SearchBoxComponent />
      <DataTable/> 
  </LeProvedore>, container)
  })
  const checkboxButton = document.querySelector("#chkOnStock")
  const dataTable = document.querySelector("#dataTable")
  expect(pretty(container.innerHTML)).toMatchSnapshot();
  expect(checkboxButton.checked).not.toBeTruthy()
  act(()=>{
    checkboxButton.dispatchEvent(new MouseEvent("click"), {bubbles: true})
  })
  expect(checkboxButton.checked).toBeTruthy()
  expect(
    pretty(dataTable.innerHTML)
  ).toMatchSnapshot()
})