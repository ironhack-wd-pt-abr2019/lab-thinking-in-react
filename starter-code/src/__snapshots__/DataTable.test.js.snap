// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Que le pueda dar click al control del checkbox 1`] = `
"<div>
  <h1>Busca tus art√≠culos</h1><input type=\\"text\\" name=\\"busqueda\\" value=\\"\\"><input type=\\"checkbox\\" id=\\"chkOnStock\\" name=\\"chkOnStock\\" value=\\"false\\">Only show products on stock
</div>
<div>
  <table id=\\"dataTable\\">
    <thead>
      <tr>
        <th>Name </th>
        <th>Price</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Football</td>
        <td>$49.99</td>
      </tr>
      <tr>
        <td>Baseball</td>
        <td>$9.99</td>
      </tr>
      <tr>
        <td>Basketball</td>
        <td>$29.99</td>
      </tr>
      <tr>
        <td>Tennis Balls</td>
        <td>$199.99</td>
      </tr>
      <tr>
        <td>iPod Touch</td>
        <td>$99.99</td>
      </tr>
      <tr>
        <td>iPhone X</td>
        <td>$399.99</td>
      </tr>
      <tr>
        <td>iPad Mini</td>
        <td>$199.99</td>
      </tr>
      <tr>
        <td>Huawei P10</td>
        <td>$199.99</td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Que le pueda dar click al control del checkbox 2`] = `
"<thead>
  <tr>
    <th>Name </th>
    <th>Price</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Football</td>
    <td>$49.99</td>
  </tr>
  <tr>
    <td>Baseball</td>
    <td>$9.99</td>
  </tr>
  <tr>
    <td>Basketball</td>
    <td>$29.99</td>
  </tr>
  <tr>
    <td>Tennis Balls</td>
    <td>$199.99</td>
  </tr>
  <tr>
    <td>iPod Touch</td>
    <td>$99.99</td>
  </tr>
  <tr>
    <td>iPhone X</td>
    <td>$399.99</td>
  </tr>
  <tr>
    <td>iPad Mini</td>
    <td>$199.99</td>
  </tr>
  <tr>
    <td>Huawei P10</td>
    <td>$199.99</td>
  </tr>
</tbody>"
`;
